pkg/utils/utils_test.go

```diff
@@ -85,10 +87,99 @@ func TestCustomizeLibertyEnv(t *testing.T) {
 	if err := verifyTests(testEnv); err != nil {
 		t.Fatalf("%v", err)
 	}
+}
+
+func TestCustomizeEnvSSO(t *testing.T) {
+	logf.SetLogger(logf.ZapLogger(true))
+	os.Setenv("WATCH_NAMESPACE", namespace)
+	svc := &openlibertyv1beta1.OpenLibertyApplicationService{Port: 8080, Type: &clusterType}
+	spec := openlibertyv1beta1.OpenLibertyApplicationSpec{Service: svc}
 
+	terminationPolicy := v1.TLSTerminationReencrypt
+	expose := true
+	spec.Env = []corev1.EnvVar{
+		{Name: "SEC_TLS_TRUSTDEFAULTCERTS", Value: "true"},
+		{Name: "SEC_IMPORT_K8S_CERTS", Value: "true"},
+	}
+	spec.Expose = &expose
+	spec.Route = &v1beta1.OpenLibertyApplicationRoute{
+		Host:        "myapp.mycompany.com",
+		Termination: &terminationPolicy,
+		Certificate: &v1beta1.Certificate{},
+	}
+	spec.SSO = &v1beta1.OpenLibertyApplicationSSO{
+		Github: &v1beta1.GithubLogin{Hostname: "github.com"},
+		OIDC: []v1beta1.OidcClient{
+			{
+				DiscoveryEndpoint: "myapp.mycompany.com",
+			},
+		},
+		Oauth2: []v1beta1.OAuth2Client{
+			{
+				ID: "custom1",
+				AuthorizationEndpoint: "specify-required-value",
+				TokenEndpoint: "specify-required-value",
+			},
+			{
+				ID: "custom2",
+				AuthorizationEndpoint: "specify-required-value",
+				TokenEndpoint: "specify-required-value",
+			},
+		},
+	}
+	data :=  map[string][]byte{
+			"github-clientId": []byte("bW9vb29vb28="),
+			"github-clientSecret": []byte("dGhlbGF1Z2hpbmdjb3c="),
+			"oidc-clientId": []byte("bW9vb29vb28="),
+			"oidc-clientSecret": []byte("dGhlbGF1Z2hpbmdjb3c="),
+	}
+	pts := &corev1.PodTemplateSpec{}
+	ssoSecret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name+"-sso",
+			Namespace: namespace,
+		},
+		Type: corev1.SecretTypeOpaque,
+		Data: data,
+	}
+
+	openliberty := createOpenLibertyApp(name, namespace, spec)
+	oputils.CustomizePodSpec(pts, openliberty)
+	CustomizeEnvSSO(pts, openliberty, ssoSecret)
+
+	podEnv := envSliceToMap(pts.Spec.Containers[0].Env, data, t)
+	tests := []Test{
+		{"Github clientid set", string(data["github-clientId"]), podEnv["SEC_SSO_GITHUB_CLIENTID"]},
+		{"Github clientSecret set", string(data["github-clientSecret"]), podEnv["SEC_SSO_GITHUB_CLIENTSECRET"]},
+		{"OIDC clientId set", string(data["oidc-clientId"]), podEnv["SEC_SSO_OIDC_CLIENTID"]},
+		{"OIDC clientSecret set", string(data["oidc-clientSecret"]), podEnv["SEC_SSO_OIDC_CLIENTSECRET"]},
+		{"Github hostname set", "github.com", podEnv["SEC_SSO_GITHUB_HOSTNAME"]},
+		{"OIDC discovery endpoint", "myapp.mycompany.com", podEnv["SEC_SSO_OIDC_DISCOVERYENDPOINT"]},
+		{"custom1 authorization endpoint", "specify-required-value", podEnv["SEC_SSO_CUSTOM1_AUTHORIZATIONENDPOINT"]},
+		{"custom1 token endpoint", "specify-required-value", podEnv["SEC_SSO_CUSTOM1_TOKENENDPOINT"]},
+		{"custom2 authorization endpoint", "specify-required-value", podEnv["SEC_SSO_CUSTOM2_AUTHORIZATIONENDPOINT"]},
+		{"custom2 token endpoint", "specify-required-value", podEnv["SEC_SSO_CUSTOM2_TOKENENDPOINT"]},
```

Comment:
Each SSO parameter is processed individually. It'll be good to verify in unit tests that ALL applicable parameters for SSO are processed correctly.